core/src/main/java/fj/data/List.java

```diff
@@ -1840,4 +1839,31 @@ private void copy() {
     @Override public String toString() {
         return Show.listShow( Show.<A>anyShow() ).show( this ).foldLeft((s, c) -> s + c, "" );
     }
+
+    /**
+     * True if the list is a singleton.  Faster than checking <code>length() == 1</code> for any
+     * list with <code>length() > 1</code>.
+     */
+	public boolean isSingle() {
+		return isNotEmpty() && tail().isEmpty();
+	}
```

Comment:
If would say `True if and only if the list has one element.  Runs in constant time.`

When you mentioned singleton I thought of the singleton design pattern.  

Not completely certain about the name `isSingle`.  I tried thinking of others: hasSizeOne, isSizeOne, isOne.  To me, none are significantly better.